demos/storybook/stories/drawer/with-basic-config.tsx

```diff
@@ -1,54 +1,57 @@
-import { Accessibility, Menu, NotificationsActive, Person, Today, Gavel, Settings } from '@material-ui/icons';
+import { Accessibility, Menu, NotificationsActive, Person, Today } from '@material-ui/icons';
 import { Drawer, DrawerBody, DrawerHeader, DrawerNavGroup, NavItem } from '@pxblue/react-components';
 import { boolean, text } from '@storybook/addon-knobs';
 import { StoryFnReactReturnType } from '@storybook/react/dist/client/preview/types';
 import React from 'react';
 import { WITH_MIN_PROPS_STORY_NAME } from '../../src/constants';
-import { DrawerStoryContext } from './util';
+import { useState } from '@storybook/addons';
+import { State, Store } from '@sambego/storybook-state';
+import { DrawerState } from './util';
 
-export const navGroupItems1: NavItem[] = [
-    {
-        title: 'Identity Management',
-        itemID: '1',
-        icon: <Person />,
-    },
-    {
-        title: 'Calendar',
-        itemID: '2',
-        icon: <Today />,
-    },
-    {
-        title: 'Accessibility',
-        itemID: '3',
-        icon: <Accessibility />,
-    },
-    {
-        title: 'Notifications',
-        itemID: '4',
-        icon: <NotificationsActive />,
-    },
-];
+const store: Store<DrawerState> = new Store<DrawerState>({
+    selected: '',
+});
 
-export const navGroupItems2: NavItem[] = [
-    {
-        title: 'Settings',
-        itemID: '2-1',
-        icon: <Settings />,
-    },
-    {
-        title: 'Legal',
-        itemID: '2-2',
-        icon: <Gavel />,
-    },
-];
+export const withBasicConfig = (): StoryFnReactReturnType => {
+    const [selected, setSelected] = useState('');
 
-export const withBasicConfig = (context: DrawerStoryContext): StoryFnReactReturnType => (
-    <Drawer open={boolean('open', true)} activeItem={context.state.selected}>
-        <DrawerHeader icon={<Menu />} title={text('title', 'Simple Drawer')} />
-        <DrawerBody>
-            <DrawerNavGroup items={navGroupItems1} />
-        </DrawerBody>
-    </Drawer>
-);
+    const navGroupItems: NavItem[] = [
+        {
+            title: 'Identity Management',
+            itemID: '1',
+            icon: <Person />,
+            onClick: () => setSelected('1'),
+        },
+        {
+            title: 'Calendar',
+            itemID: '2',
+            icon: <Today />,
+            onClick: () => setSelected('2'),
+        },
+        {
+            title: 'Accessibility',
+            itemID: '3',
+            icon: <Accessibility />,
+            onClick: () => setSelected('3'),
+        },
+        {
+            title: 'Notifications',
+            itemID: '4',
+            icon: <NotificationsActive />,
+            onClick: () => setSelected('4'),
+        },
+    ];
```

Comment:
You can make these menu items a shared function so you don't have to copy/paste/redefine them in every story.

Reply:
I thought the idea is so that the users can copy from our code directly.

If these are shared then we need to include a comment that links to where this piece of code is on GitHub

Reply:
The previous implementation also used shared definitions of the menu items across stories. The primary use case of storybook is not to be able to copy and paste code into an application and have it work (unless they are building a storybook app) — it would be nice, but the way storybook works right now, I don't think it's possible (think about all of the knob code and the fact that the examples are not even react components, but `StoryFnReactReturnType`). We were hoping the storybook 6 story source plugin would give us more hope for being able to copy/paste from storybook, but that's still unknown.

The _design pattern_ examples' primary use case is for copy/pastabiity.

Reply:
I'm fine either way depending on what is more likely — are we more likely to want to make changes to the drawer items on a per-story basis (favor defining them individually) or are we more likely to want to make a change to the items that will be reflected in all of the stories (favor shared definition).